### PR TITLE
Explain about the automated tests on Travis

### PR DESCRIPTION
**Pull request by nodakai**
in order to advise contributors to locally test their patches.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -88,6 +88,16 @@ With that in mind, the steps for adding a new API are:
 4. Wait for CI to pass, fixing errors.
 5. Wait for a merge!
 
+### Test before you commit
+
+We have two automated tests running on [Travis](https://travis-ci.org/rust-lang/libc):
+
+1. `libc-test`
+  - `cd libc-test && cargo clean && cargo run`
```
Comment (alexcrichton) [inline README.md]: In theory this should just be `cargo run --manifest-path libc-test/Cargo.toml`, how come the `cargo clean` is here?

**Comment by alexcrichton** [comment]
Thanks! Definitely seems like good information to have.

File: README.md (hunk 1 above)
Comment (nodakai) [inline README.md]: `cargo clean` is for making sure `all.c` is regenerated according to the latest code of the `libc` crate.

```
$ cd libc-test; cargo build
$ echo 'extern { pub fn xyzzy() -> ::c_int; }' >> ../src/unix/mod.rs
$ cargo run
   Compiling libc v0.2.9 (file:///tmp/libc/libc-test)
   Compiling libc-test v0.1.0 (file:///tmp/libc/libc-test)
     Running `target/debug/libc-test`
RUNNING ALL TESTS
PASSED 6086 tests
$ cargo clean; cargo run                                                                                                                                                                                                                 (git)-master 
   Compiling winapi v0.2.6
   Compiling winapi-build v0.1.1
   Compiling gcc v0.3.25
   Compiling libc v0.2.8
   Compiling unicode-xid v0.0.3
   Compiling bitflags v0.3.3
   Compiling rustc-serialize v0.3.18
   Compiling libc v0.1.12
   Compiling libc v0.2.9 (file:///tmp/libc/libc-test)
   Compiling kernel32-sys v0.2.1
   Compiling term v0.2.14
   Compiling log v0.3.5
   Compiling syntex_syntax v0.19.1
   Compiling ctest v0.1.0 (https://github.com/alexcrichton/ctest#50ac771a)
   Compiling libc-test v0.1.0 (file:///tmp/libc/libc-test)
failed to run custom build command for `libc-test v0.1.0 (file:///tmp/libc/libc-test)`
Process didn't exit successfully: `/tmp/libc/libc-test/target/debug/build/libc-test-475f1c4bf6cbc262/build-script-build` (exit code: 101)
...snip...
--- stderr
/tmp/libc/libc-test/target/debug/build/libc-test-475f1c4bf6cbc262/out/all.c: In function ‘__test_fn_xyzzy’:
/tmp/libc/libc-test/target/debug/build/libc-test-475f1c4bf6cbc262/out/all.c:12758:24: error: ‘xyzzy’ undeclared (first use in this function)
                 return xyzzy;
                        ^
/tmp/libc/libc-test/target/debug/build/libc-test-475f1c4bf6cbc262/out/all.c:12758:24: note: each undeclared identifier is reported only once for each function it appears in
/tmp/libc/libc-test/target/debug/build/libc-test-475f1c4bf6cbc262/out/all.c:12759:13: error: control reaches end of non-void function [-Werror=return-type]
             }
             ^
...snip...
```

Comment (alexcrichton) [inline README.md]: I think that was fixed in a more recent version of `ctest`, can you try updating that dependency for ctest?

Comment (nodakai) [inline README.md]: Isn't it not enough to run `rm -rf libc; rm -rf ~/.cargo; git clone https://github.com/rust-lang/libc.git` ?

Comment (alexcrichton) [inline README.md]: Nah we check in a lock file here so you'll need to run `cargo update`

Comment (nodakai) [inline README.md]: OK, confirmed, but

1. Please update the lock file on your side. In a sense you chose not to release the fix on `rerun-if-changed` by distributing the lock file.
2. I found the semantic version of a Git dependency confusing. I filed rust-lang/cargo#2586

**Comment by alexcrichton** [comment]
> In a sense you chose not to release the fix on rerun-if-changed by distributing the lock file.

Or rather I chose to have stability over gaining new features asap :)

Do you want to check in the upgrade to `Cargo.lock` as part of this PR as well?

**Comment by nodakai** [comment]
- https://travis-ci.org/rust-lang/libc/jobs/125987669

```
+curl https://static.rust-lang.org/rustup.sh

++rustc --print sysroot

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current

                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0+sh -s -- --add-target=x86_64-apple-ios --disable-sudo -y --prefix=/Users/travis/rust

100 60459  100 60459    0     0   107k      0 --:--:-- --:--:-- --:--:--  107k

rustup: assert_nz url is empty!
```

**Pull request merged**